modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/MergeJoinNode.java

```diff
@@ -989,10 +991,12 @@ protected void join() throws Exception {
                     } else {
                         if (left == null) {
                             if (waitingLeft == NOT_WAITING) {
+                                rightIdx = 0;
```

Comment:
Should we introduce a methods like
```
        /** */
        private void clearRightMaterialization() {
            rightMaterialization = null;
            drainMaterialization = false;
            rightIdx = 0;
        }
```
?
Sometimes we do
```
   rightIdx = 0;
   rightMaterialization = null;
```
without
`drainMaterialization = false`